File: src/models/track.ts
export interface Track {
  id: string;
  title: string;
  artist: string;
  album: string;
  duration: number; // in seconds
  url: string;
  coverUrl: string;
  bpm?: number;     // Beats Per Minute for musicians
  key?: string;     // Musical Key (e.g., "Am", "C#", "G") for musicians
}

File: src-tauri/src/models/track.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Track {
    pub id: String,
    pub title: String,
    pub artist: String,
    pub album: String,
    pub duration: f64,
    pub url: String,
    #[serde(rename = "coverUrl")]
    pub cover_url: String,
    pub bpm: Option<u32>,
    pub key: Option<String>,
}
